interfaces/matlab_experimental/Utility/ctLib.m

```diff
@@ -1,55 +1,69 @@
 function str = ctLib(libDir)
     % Return name of Cantera Shared Library depending on OS.
+    function score = versionScore(libName)
+        % Pattern to match version numbers (x.y.z, x.y, or x)
+        % Matches sequences of digits separated by dots
+        pattern = '(\d+(?:\.\d+){0,2})(?=\.(?:dylib|so)|$)';
+
+        % Find the version number
+        tokens = regexp(libName, pattern, 'tokens', 'once');
+        if ~isempty(tokens)
+            verStr = tokens{1};
+        else
+            score = -1;
+            return;
+        end
+
+        % Parse version string into comparable score
+        parts = sscanf(verStr, '%d.%d.%d');
+        parts(end+1:3) = 0;  % Pad to length 3
+        score = parts(1)*1e6 + parts(2)*1e3 + parts(3);
+    end
+
     if ispc
-        str = libDir + "/cantera_shared.dll";
+        if isempty(dir(fullfile(libDir, "cantera_shared.dll")))
+            error('No matching shared libraries found in %s', libDir);
+        end
+        str = string(fullfile(libDir, "cantera_shared.dll"));
     elseif ismac
-        if ~isempty(dir(fullfile(libDir, 'libcantera.dylib')))
-            baseName = 'libcantera';
-        elseif ~isempty(dir(fullfile(libDir, 'libcantera_shared.dylib')))
+        if ~isempty(dir(fullfile(libDir, 'libcantera_shared*.dylib')))
             baseName = 'libcantera_shared';
+        elseif ~isempty(dir(fullfile(libDir, 'libcantera*.dylib')))
+            baseName = 'libcantera';
         else
             error('No matching shared libraries found in %s', libDir);
         end
         files = dir(fullfile(libDir, [baseName, '*.dylib']));
-        versioned = {};
-        for k = 1:numel(files)
-            name = files(k).name;
-            tokens = regexp(name, [baseName,'\.([0-9]+(\.[0-9]+)*)\.dylib$'], 'tokens', 'once');
-            if ~isempty(tokens)
-                versioned{end+1} = name;
-            end
-        end
-        if isempty(versioned)
-            warning('No versioned library found, falling back to unversioned');
-            str = libDir + baseName + ".dylib";
+
+        scores = arrayfun(@(f) versionScore(f.name), files);
+        hasVersion = scores > 0;
+        if any(hasVersion)
+            [~, idx] = max(scores);
+            str = string(fullfile(libDir, files(idx).name));
         else
```

Comment:
I realize this was a preexisitng condition, but I think this mechanism of interacting with the versioned shared libraries is completely ineffective. There is nothing to make sure that the Matlab toolbox corresponds to the version of `libcantera` that is found here. In the absence of a more robust mechanism for this, I would suggest just taking the simple approach and linking to the unversioned library name.

I also have no idea what a version "score" is, but that's sort of a moot point compared to the larger issue here.